jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/kubernetes/KubernetesClientUtil.java

```diff
@@ -155,34 +153,21 @@ protected static String getPodCondition(Pod pod) {
         return "";
     }
 
-    public static GenericCustomResource doGetCustomResource(KubernetesClient kubernetesClient, CustomResourceDefinitionContext crdContext, String namespace, String name) {
+    public static GenericKubernetesResource doGetCustomResource(KubernetesClient kubernetesClient, String apiVersion, String kind, String namespace, String name) {
         try {
-            return Serialization.jsonMapper().convertValue(
-                    kubernetesClient.customResource(crdContext).inNamespace(namespace).withName(name).get(),
-                    GenericCustomResource.class
-            );
+            return kubernetesClient.genericKubernetesResources(apiVersion, kind).inNamespace(namespace).withName(name).get();
         } catch (Exception exception) { // Not found exception
             return null;
         }
     }
 
-    public static void doDeleteAndWait(KubernetesClient kubernetesClient, CustomResourceDefinitionContext context,
+    public static void doDeleteAndWait(KubernetesClient kubernetesClient, String apiVersion, String kind,
```

Comment:
I'm not sure if it might make sense now to change the method signature to:

```java
    public static void doDeleteAndWait(KubernetesClient kubernetesClient, GenericKubernetesResource resource,
                                       String namespace, long seconds) {
```